intercept with --local-only and --docker-run crashes

**Describe the bug**

When running an intercept with `--local-only` and `--docker-run` telepresence crashes with the following error:

```
telepresence intercept my-deployment --local-only --docker-run -- alpine:latest
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0x101774fd2]

goroutine 1 [running]:
github.com/telepresenceio/telepresence/v2/pkg/client/cli/intercept.(*state).startInDocker(0xc000206050, {0x1027a2ac8, 0xc000a58b70}, {0xc0001a1510, 0xd}, {0xc00005e140, 0x43}, {0xc0005a59f0, 0x1, 0x8})
        github.com/telepresenceio/telepresence/v2@v2.13.1/pkg/client/cli/intercept/docker_run.go:200 +0x952
github.com/telepresenceio/telepresence/v2/pkg/client/cli/intercept.runCommand({0x1027a8438, 0xc00037c030}, {0x1027a2ac8, 0xc00056d230})
        github.com/telepresenceio/telepresence/v2@v2.13.1/pkg/client/cli/intercept/state.go:387 +0x497
github.com/telepresenceio/telepresence/v2/pkg/client.WithEnsuredState[...]({0x1027a2ac8, 0xc00056d230}, {0x1027a8438, 0xc00037c030}, 0x10001bb53, 0x10251b7f0?, 0x10251b7e0)
        github.com/telepresenceio/telepresence/v2@v2.13.1/pkg/client/ensured_state.go:39 +0x1b9
github.com/telepresenceio/telepresence/v2/pkg/client/cli/intercept.Run({0x1027a2ac8, 0xc00056d230}, {0x1027a8438?, 0xc00037c030?})
        github.com/telepresenceio/telepresence/v2@v2.13.1/pkg/client/cli/intercept/state.go:202 +0x170
github.com/datawire/telepresence-pro/pkg/client/cli/intercept.Run({0x1027a2ac8, 0xc00056d230}, {0x1027a8438?, 0xc00037c030?}, {0xc00032ac00, 0x24})
        github.com/datawire/telepresence-pro/pkg/client/cli/intercept/state.go:51 +0xbc
github.com/datawire/telepresence-pro/pkg/client/cli/intercept.(*Command).Run(0xc000384d80, 0xc000a0a900, {0xc0005a59e0, 0x2, 0x9})
        github.com/datawire/telepresence-pro/pkg/client/cli/intercept/intercept.go:238 +0x41e
github.com/spf13/cobra.(*Command).execute(0xc000a0a900, {0xc0005a5950, 0x9, 0x9})
        github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc000a0bb00)
        github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/telepresenceio/telepresence/v2/pkg/client/cli/output.Execute(0x1027a2ac8?)
        github.com/telepresenceio/telepresence/v2@v2.13.1/pkg/client/cli/output/output.go:114 +0x3d
github.com/telepresenceio/telepresence/v2/pkg/client/cli.Main({0x1027a2ac8, 0xc0009e1020})
        github.com/telepresenceio/telepresence/v2@v2.13.1/pkg/client/cli/main.go:68 +0x30f
main.main()
        github.com/datawire/telepresence-pro/cmd/telepresence/main.go:11 +0x2c
```

**To Reproduce**
Steps to reproduce the behavior:
1. Run `telepresence intercept my-deployment --local-only --docker-run -- alpine:latest`
2. telepresence crash

**Versions (please complete the following information):**

```
> telepresence version         
Client             : v2.13.1
Daemon in container: v2.13.1
Traffic Manager    : v2.13.1
```

Running on MacOS Monterey and GKE

**Additional context**

Looking at the code it seems to be failing at:

https://github.com/telepresenceio/telepresence/blob/6855291201400a1b5214ecaccbde9a1432a9c11c/pkg/client/cli/intercept/docker_run.go#L200

I think `s.info` is `null` when `--local-only` flag is set because of the early return at 

https://github.com/telepresenceio/telepresence/blob/6855291201400a1b5214ecaccbde9a1432a9c11c/pkg/client/cli/intercept/state.go#L260-L262

Note: I've tried to add `--mount false` since the code is crashing when trying to create the mounts but this doesn't work either as I get this error instead 

```
telepresence intercept: error: a local-only intercept cannot have mounts
```

Which is a bit misleading since I'm actually saying I don't want mounts.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,28 +1,45 @@
 # Changelog
 
+### 2.13.3 (TBD)
+- Bugfix: Running `telepresence intercept --local-only --docker-run` no longer results in a panic.
+  Ticket [3171](https://github.com/telepresenceio/telepresence/issues/3171).
+
+- Bugfix: Running `telepresence intercept --local-only --mount false` no longer results in an incorrect error message
+  saying "a local-only intercept cannot have mounts".
+  Ticket [3171](https://github.com/telepresenceio/telepresence/issues/3171).
+
 ### 2.13.2 (May 12, 2023)
 - Bugfix: Replaced `/` characters with a `-` when the authenticator service creates the kubeconfig in the Telepresence cache.
+  PR [3167](https://github.com/telepresenceio/telepresence/issues/3167).
 
 - Feature: Configurable strategy (`auto`, `powershell`. or `registry`) to set the global DNS search path on Windows. Default
   is `auto` which means try `powershell` first, and if it fails, fall back to `registry`.
+  Ticket [3152](https://github.com/telepresenceio/telepresence/issues/3152).
 
 - Feature: The timeout for the traffic manager to wait for traffic agent to arrive can
   now be configured in the `values.yaml` file using `timeouts.agentArrival`. The default
   timeout is still 30 seconds.
+  PR [3148](https://github.com/telepresenceio/telepresence/issues/3148).
 
 - Bugfix: The automatic discovery of a local container based cluster (minikube or kind) used when the
   Telepresence daemon runs in a container, now works on macOS and Windows, and with different profiles,
   ports, and cluster names
+  PR [3165](https://github.com/telepresenceio/telepresence/issues/3165).
 
 - Bugfix: FTP Stability improvements. Multiple simultaneous intercepts can transfer large files in bidirectionally and in parallel.
+  PR [3157](https://github.com/telepresenceio/telepresence/issues/3157).
 
 - Bugfix: Pods using persistent volumes no longer causes timeouts when intercepted.
 
 - Bugfix: Ensure that `telepresence connect` succeeds even though DNS isn't configured correctly.
+  Ticket [3143](https://github.com/telepresenceio/telepresence/issues/3143).
+  PR [3154](https://github.com/telepresenceio/telepresence/issues/3154).
 
 - Bugfix: The traffic-manager would sometimes panic with a "close of closed channel" message and exit.
+  PR [3160](https://github.com/telepresenceio/telepresence/issues/3160).
 
 - Bugfix: The traffic-manager would sometimes panic and exit after some time due to a type cast panic.
+  Ticket [3149](https://github.com/telepresenceio/telepresence/issues/3149).
 
 ### 2.13.1 (April 20, 2023)
 

--- a/pkg/client/cli/intercept/command.go
+++ b/pkg/client/cli/intercept/command.go
@@ -120,7 +120,9 @@ func (a *Command) Validate(cmd *cobra.Command, positional []string) error {
 			return errcat.User.New("a local-only intercept cannot have a port")
 		}
 		if cmd.Flag("mount").Changed {
-			return errcat.User.New("a local-only intercept cannot have mounts")
+			if doMount, _ := a.GetMountPoint(); doMount {
+				return errcat.User.New("a local-only intercept cannot have mounts")
+			}
 		}
 		return nil
 	}

--- a/pkg/client/cli/intercept/docker_run.go
+++ b/pkg/client/cli/intercept/docker_run.go
@@ -196,7 +196,7 @@ func (s *state) startInDocker(ctx context.Context, daemonName, envFile string, a
 		if !set {
 			ourArgs = append(ourArgs, "--rm")
 		}
-		if !s.mountDisabled {
+		if !(s.mountDisabled || s.info == nil) {
 			m := s.info.Mount
 			if m != nil {
 				if err := docker.EnsureVolumePlugin(ctx); err != nil {

--- a/pkg/client/cli/intercept/state.go
+++ b/pkg/client/cli/intercept/state.go
@@ -94,6 +94,7 @@ func (s *state) CreateRequest(ctx context.Context) (*connector.CreateInterceptRe
 
 	if s.AgentName == "" {
 		// local-only
+		s.mountDisabled = true
 		return ir, nil
 	}
 

